### DISCUSSION TITLE
Sub-Application routes are not shown in root routes

### DISCUSSION THREAD

**Discussion by ierezell** [fastapi/fastapi#8806] [Questions]
### First Check

- [X] I added a very descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
# From the docs : https://fastapi.tiangolo.com/advanced/sub-applications/
from fastapi import FastAPI

app = FastAPI()


@app.get("/app")
def read_main():
    return {"message": "Hello World from main app"}


subapi = FastAPI()


@subapi.get("/sub")
def read_sub():
    return {"message": "Hello World from sub API"}


app.mount("/subapi", subapi)
```

### Description

The sub Apis routes are not shown in the root docs. 

I'm adding dynamically multiple sub Apis in my root app and it's hard to keep track of which is mounted and on which names. 

Is there a way to show at least the name of the sub API route in the root app?

Example Docs at the root : 
```
/app # (exists) Description of  /app route in root
/subapi # (Do not exists) Description of /subapi (at least to know another app is mounted)
```

Nother Exemple of Docs at the root : 
```
/app # Description of  /app route in root
# SubApi Tag to know there is another app 
/docs # Know that there is docs inside subapp
/sub  # Know that there is sub route inside sub app
```

### Operating System

Linux

### Operating System Details

Archlinux Kernel 5.13.10

### FastAPI Version

0.66.0

### Python Version

3.9.6

### Additional Context

Thanks for this beautiful library !

**Comment by kauedg** [comment]
I've been using subapps to manage different API versions and these versions are listed by a "information" endpoint. 

```python
import starlette
from fastapi import FastAPI, Request
from versions import v0, v1

class ApiInfo(object):
    """docstring for ApiInfo."""

    def __init__(
        self,
        app: FastAPI,
        request: Request):

        super(ApiInfo, self).__init__()

        self.url = str(request.url)

        self.autodocs = {
            'swagger': str(request.url) + app.docs_url.strip("/"),
            'redoc': str(request.url) + app.redoc_url.strip("/"),
        }

        # Gera a lista de versões
        self.versions = {}
        for version in app.routes:
            if type(versions) is starlette.routing.Mount:
                self.versions[version.name] =  str(request.url) + version.name


app = FastAPI(
    title="Root app",
)

app.mount("/v0", v0.app, "v0")
app.mount("/v1", v1.app, "v1")

@app.get(
   '/',
   summary='API Information',
   tags=['API'],
)
def info(
   request: Request,
):
   apiInfo = ApiInfo(app, request)

   apiInfo.versions['stable'] = apiInfo.versions['v0']
   apiInfo.versions['develop'] = apiInfo.versions['v1']
   # whatever other label/mount name you want
   
   return apiInfo.__dict__

```

It will return something like this

```json
{
  "url": "http://192.168.0.20:8686/",
  "versions": {
    "v0": "http://192.168.0.20:8686/v0",
    "v1": "http://192.168.0.20:8686/v1",
    "stable": "http://192.168.0.20:8686/v0"
    "develop": "http://192.168.0.20:8686/v1"
  },
  "autodocs": {
    "swagger": "http://192.168.0.20:8686/docs",
    "redoc": "http://192.168.0.20:8686/redoc"
  }
}
```

**Comment by ierezell** [comment]
Hi @kauedg, 

Thanks a lot for this answer! 

with 
```python
for version in app.routes:
    if type(versions) is starlette.routing.Mount:
        # url + version.name
```
I will be able to at least add it to the docstring of the root app to list and link to sub apps... 

However, I would still like to have an automatic way to mount and show them in the UI, I don't know if it's a feature that the framework would like to support. 

Have a great day

**Comment by kauedg** [comment]
Maybe you could list the subapps in the "servers" field from the OpenAPI spec?

https://swagger.io/specification/#server-object
https://github.com/tiangolo/fastapi/blob/b8c4149e89d1c97d204ac3f965e6144e3dc126a9/fastapi/applications.py#L42

That would still be kind of a manual task to add the subapps to the list but eliminates the need for an HTTP request. And I see potential for a new feature.

Edit: this is not fit for your needs as the OpenAPI's "servers" object only redirects the request to a different server with the same defined endpoints. 

The OpenAPI spec has no support for such use case.

**Comment by frerikandriessen** [comment]
I would also like to have an overview of all the mounted routes on the root docs. Is this something FastAPI is willing to accept a PR for?

**Comment by vkoivula** [comment]
Would make sense to be able to configure that the documentation for each subapi would appear under the /docs (or whatever you set "the OG" root directory for docs).
